src/main/java/org/osc/manager/rest/server/api/SecurityApis.java

```diff
@@ -0,0 +1,173 @@
+        LOG.info("Listing security group ids'");
+        List<? extends ManagerSecurityGroupElement> securityGroups = this.sgApi.getSecurityGroupList();
+        List<String> sgList = new ArrayList<String>();
+        if (securityGroups.isEmpty()) {
+            return sgList;
+        }
+        sgList = securityGroups.stream().map(ManagerSecurityGroupElement::getSGId).collect(Collectors.toList());
+        return sgList;
+    }
+
+    @Path("/{sgId}")
+    @GET
+    public SecurityGroupEntity getSecurityGroup(@PathParam("sgId") Long sgId) throws Exception {
+        LOG.info(String.format("Getting the security group for id %s ", "" + Long.toString(sgId)));
+        return this.sgApi.getSecurityGroup(Long.toString(sgId));
+    }
+
+    @Path("/{sgId}/sgIntf")
+    @POST
+    public String createSecurityGroupInterface(@PathParam("sgId") Long sgId, SecurityGroupInterfaceEntity entity)
+            throws Exception {
+        LOG.info(String.format("Creating the security group interface with name %s ", "" + entity.getName()));
+        SecurityGroupEntity sgElement = new SecurityGroupEntity();
+        sgElement.setId(sgId);
+        entity.setSecurityGroup(sgElement);
+        return this.sgiApi.createSecurityGroupInterface(entity);
+    }
+
+    @Path("/{sgId}/sgIntf/{sgIntfId}")
+    @PUT
+    public SecurityGroupInterfaceElement updateSecurityGroupInterface(@PathParam("sgId") Long sgId,
+            @PathParam("sgIntfId") Long sgIntfId, SecurityGroupInterfaceEntity entity) throws Exception {
+        LOG.info(String.format("Updating the security group interface with sgid %s ; sginterfaceid %s",
+                "" + Long.toString(sgId), "" + Long.toString(sgIntfId)));
+        SecurityGroupEntity sgElement = new SecurityGroupEntity();
+        sgElement.setId(sgId);
+        entity.setSecurityGroup(sgElement);
+        entity.setId(sgIntfId);
+        this.sgiApi.updateSecurityGroupInterface(entity);
+        return entity;
+    }
+
+    @Path("/{sgId}/sgIntf/{sgIntfId}")
+    @DELETE
+    public void deleteSecurityGroupInterface(@PathParam("sgId") Long sgId, @PathParam("sgIntfId") Long sgIntfId)
+            throws Exception {
+        LOG.info(String.format("Deleting the security group interface with sgid %s ; sginterfaceid %s",
+                "" + Long.toString(sgId), "" + Long.toString(sgIntfId)));
+        this.txControl.required(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                SecurityApis.this.sgiApi.deleteSecurityGroupInterface(Long.toString(sgIntfId));
+                return null;
+            }
+        });
+    }
+
+    @Path("/{sgId}/sgIntf")
+    @GET
+    public List<String> getSecurityGroupInterfaceIds() throws Exception {
+        LOG.info("Listing security group interface ids'");
+        List<? extends ManagerSecurityGroupInterfaceElement> sgiElements = this.sgiApi.listSecurityGroupInterfaces();
```

Comment:
This is not filtering by SG id, it is returning all the interfaces.  You likely need to add a new DB query to use the SG id as well, the SDK implementation may not have that query already.

Reply:
We have 1-to-1 relation between SG and SGI and listing all means only one SGI for each SG. 
Then this will return only one SGI for SG.
I will fix this way.